src/compiler/crystal/macros/interpreter.cr

```diff
@@ -371,15 +371,17 @@ module Crystal
     end
 
     def resolve(node : Path)
+      resolve?(node) || node.raise_undefined_constant(@path_lookup)
+    end
+
+    def resolve?(node : Path)
       if node.names.size == 1 && (match = @free_vars.try &.[node.names.first]?)
         matched_type = match
       else
         matched_type = @path_lookup.lookup_path(node)
       end
 
-      unless matched_type
-        node.raise_undefined_constant(@path_lookup)
-      end
+      return unless matched_type
```

Comment:
Leaving out nil seems to be the more prevalent style so far, but there's quite a bit with nil too, and I don't mind either way. If we want to be consistent with `return nil (if|unless)`, then maybe we should disallow not specifying it in the compiler (or formatter).

```
$ grep -ER 'return (unless|if)' src | wc -l
264
$ grep -ER 'return nil (unless|if)' src | wc -l
81
```